Added feature to stop generating invoices after contract goes in overdue

agb80 opened the pull request.

Commit:
The saleperson is set like the user instead of administrator

Commit:
When we are working in a environment multicompany, and it is generated an invoice the invoice created has the taxes of all the compamies instead only one

agb80 committed:
Merge pull request #3 from OpenPymeMx/taxes

When we are working in a environment multicompany, and it is generated a...

agb80 committed:
Merge pull request #2 from OpenPymeMx/invoice_owner

The saleperson is set like the user instead of administrator

Commit:
feat(reccurring contracts): stop to create invoices from a specific contract

before create a new invoice, fist review if are there  opened invoices with a past dateline from the
specific contract, if there are invoices with this conditions then the contract past to a exception
state and the invoice creation stops

agb80 committed:
feat(account_analytic_analysis_recurret): add option to stop generating invoices after overdue

When a contract got in overdue the invoices contine being generated causing lot of confusion for end
users.

agb80 committed:
feat(reccurring contracts): stop to create invoices from a specific contract

before create a new invoice, fist review if are there opened invoices with a past dateline from the
specific contract, if there are invoices with this conditions then the contract past to a exception
state and the invoice creation stops

Commit:
fix(account_analytic_account): convert frozen dict into a normal dictionary

Commit:
Merge pull request #7 from OpenPymeMx/openpyme7.0

fix(account_analytic_account): convert frozen dict into a normal dict…

dreispt commented:
@agb80 I think the right place for this module is OCA/contract. 
Do you agree @pedrobaeza ?

pedrobaeza commented:
@dreispt I agree, but remember that the contract repo was created after 7.0, and this module still resides here for 7.0 version. If we move the module, existing installations will break. That's why we keep it here.

dreispt commented:
@pedrobaeza Sorry, I didn't notice it was a 7.0 module. Agreed then.

dreispt commented on account_analytic_analysis_recurring/account_analytic_analysis_recurring.py:
```diff
@@ -66,7 +66,7 @@ def product_id_change(
             partner_id=False, price_unit=False, pricelist_id=False,
             company_id=None, context=None):
-        context = context or {}
+        context = dict(context or {})
```
I don't see the point for this.

dreispt commented on account_analytic_analysis_recurring/account_analytic_analysis_recurring.py:
```diff
@@ -123,6 +123,16 @@ class AccountAnalyticAccount(orm.Model):
+             ('overdue', 'Overdue'),
+             ('close', 'Closed'),
+             ('cancelled', 'Cancelled')], 'Status', required=True,
+            track_visibility='onchange'
```
It would be nice if you could add comments identifying what this redefinition changes from the standard.

dreispt commented:
Question: shouldn't we have a flag in the Contract to allow avoiding specific contracts to have invoicing suspended? There are always special cases...

dreispt commented:
Travis build fails because of `analytic_hours_block` :confused:

dreispt commented:
...but tests pass on module [unit test](https://travis-ci.org/OCA/project/jobs/139804342#L2055) :smile:

Commit:
fix(account_analytic_account): select correct tax in case if we had more than one company

at an multicompany environment it is possible to have more than only once equal tax per company then
this change filters the taxes per company

Commit:
Merge pull request #8 from OpenPymeMx/update_7.0

fix(account_analytic_account): select correct tax in case if we had m…

dreispt commented on account_analytic_analysis_recurring/account_analytic_analysis_recurring.py:
```diff
@@ -166,7 +166,10 @@ def _prepare_invoice_line(self, cr, uid, line, invoice_id, context=None):
-        taxes = product.taxes_id or False
+        taxes = [
+            x for x in product.taxes_id if
+            x.company_id.id == context['force_company']
```
What if no `force_company` key is in the context?

Commit:
fix(account.analytic.account): use continue instaed of break

Commit:
Merge pull request #9 from OpenPymeMx/fix_branch

fix(account.analytic.account): use continue instaed of break

pedrobaeza commented:
Comments hasn't been attended for a long time, so I close.
